apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/consumer/MultipleChannelsConsumer.java

```diff
@@ -43,10 +46,11 @@ public MultipleChannelsConsumer(String threadName, long consumeCycle) {
     public void run() {
         running = true;
 
+        final List consumeList = new ArrayList(2000);
         while (running) {
             boolean hasData = false;
             for (Group target : consumeTargets) {
-                hasData = hasData || consume(target);
+                hasData |= consume(target, consumeList);
```

Comment:
You change `||` to `|`

Reply:
They are the same here, right?

Reply:
> They are the same here, right?

Not realy, it's `short circuit` operation (`consume` won't be called if `hasData == true`) before your changes and the `consume` would always be called after your changes

Reply:
> Not realy, it's short circuit operation (consume won't be called if hasData == true) before your changes and the consume would always be called after your changes

Let's make it clear, it should consume even hasData. If old codes didn't, that is a bug by me  :( I was not expecting this before. 
Let's remove the `|` and `||`, change it to normal codes 
```java
boolean consumeResult = consume(target);
hasData = hasData || consumeResult;
```

Reply:
I got it.

Reply:
Checked. Old codes did `short circuit`, which is wrong. 
We should consume all group in one loop
I think this will improve performance.

@kezhenxu94 Thanks for pointing this out.

Reply:
I have a test, `||` is shor circuit.